src/Token/AccessToken.php

```diff
@@ -76,11 +86,25 @@ public static function resetTimeNow()
     }
 
     /**
-     * @return int
+     * @inheritdoc
+     */
+    public function setClock(Clock $clock)
+    {
+        $this->clock = $clock;
+    }
+
+    /**
+     * @inheritdoc
      */
     public function getTimeNow()
     {
-        return self::$timeNow ? self::$timeNow : time();
+        if (self::$timeNow) {
+            return self::$timeNow;
+        } elseif (isset($this->clock)) {
+            return $this->clock->now()->getTimestamp();
+        } else {
+            return time();
```

Comment:
If you implement a `getClock(): Clock` method that must always return a Clock, then there's no need to check for `isset($this->clock)`. Instead, you can do:

``` php
if (self::$timeNow) {
    return self::$timeNow;
}

return $this->getClock()->now()->getTimestamp();
```

Reply:
I think it would be better for `self::$timeNow` to be a `DateTimeImmutable`, rather than an integer.